octopusdeploy/schema_runbook.go

```diff
@@ -0,0 +1,182 @@
+package octopusdeploy
+
+import (
+	"context"
+	"fmt"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/runbooks"
+
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/core"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func expandRunbook(ctx context.Context, d *schema.ResourceData) *runbooks.Runbook {
+	name := d.Get("name").(string)
+	projectId := d.Get("project_id").(string)
+
+	runbook := runbooks.NewRunbook(name, projectId)
+	runbook.ID = d.Id()
+
+	if v, ok := d.GetOk("description"); ok {
+		runbook.Description = v.(string)
+	}
+
+	if v, ok := d.GetOk("runbook_process_id"); ok {
+		runbook.RunbookProcessID = v.(string)
+	}
+
+	if v, ok := d.GetOk("published_runbook_snapshot_id"); ok {
+		runbook.PublishedRunbookSnapshotID = v.(string)
+	}
+
+	if v, ok := d.GetOk("space_id"); ok {
+		runbook.SpaceID = v.(string)
+	}
+
+	if v, ok := d.GetOk("multi_tenancy_mode"); ok {
+		runbook.MultiTenancyMode = core.TenantedDeploymentMode(v.(string))
+	}
+
+	if v, ok := d.GetOk("connectivity_policy"); ok {
+		runbook.ConnectivityPolicy = expandConnectivityPolicy(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("environment_scope"); ok {
+		runbook.EnvironmentScope = v.(string)
+	}
+
+	if v, ok := d.GetOk("environments"); ok {
+		runbook.Environments = getSliceFromTerraformTypeList(v)
+	}
+
+	if v, ok := d.GetOk("default_guided_failure_mode"); ok {
+		runbook.DefaultGuidedFailureMode = v.(string)
+	}
+
+	if v, ok := d.GetOk("retention_policy"); ok {
+		runbook.RunRetentionPolicy = expandRunbookRetentionPolicy(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("force_package_download"); ok {
+		runbook.ForcePackageDownload = v.(bool)
+	}
+
+	return runbook
+}
+
+func getRunbookSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"id": getIDSchema(),
+		"name": {
+			Description:      "The name of the project in Octopus Deploy. This name must be unique.",
+			Required:         true,
+			Type:             schema.TypeString,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+		},
+		"description": {
+			Computed:    true,
```

Comment:
I don't think this should be computed?

Reply:
It may not need to be computed TBH. I've kept the same options from `schema_project.go` for consistency where possible, and that listed the `description` field as computed.

Reply:
yeah, it feels wrong to me, @jbristowe thoughts?